### ISSUE TITLE
Please don't `wsl --update` from Docker Desktop (without asking)

### ISSUE THREAD

**Issue by rfay** [docker/for-win#14314]
### Description

In recent versions of Docker Desktop, it has been doing an un-asked-for `wsl --update`

We don't expect an application to be changing the basic configuration of our computer. Please don't do that.

If you think the wsl version is inadequate for Docker Desktop, just report that, and tell people what to do. 

Docker Desktop does not own WSL, and it's not the only thing that may be using it. Please don't do the upgrade without asking.

### Reproduce

Run Docker Desktop with WSL2

### Expected behavior

Don't change configuration of machine without permission

### docker version

```bash
Any version with WSL2
```

### docker info

```bash
Any recent version
```

### Diagnostics ID

Doesn't need it

### Additional Info

Thank you!

**Comment by davejdyer** [comment]
I concur 100% with this, and in my case, it's now causing the install/startup to fail because it's unable to run that command on my system. It pops up an admin user acceptance black screen prompt over and over until I finally tell it "no, you can't access 'helper whatever' and it throws an error saying it couldn't run docker.admin --update-wsl or whatever. My WSL is and has been on the latest version and everything ran great until the latest update of Docker Desktop, but it apparently can't handle that fact now. I should have saved some logs to add here, but I got so frustrated that it's all gone from my system, and I'm going to wait before using it again; maybe go with Rancher Desktop for now. I've been fighting with it for two days after I made the mistake of letting it update itself when it promptly crashed in the middle of the update and left itself uninstalled. Trying to install the latest after a full cleanup has made no difference. It's one error after another.

**Comment by hrabatom** [comment]
I agree with the request. It's really annoying to get the admin prompt on each logon, so I need to enter the credentials again! (I usually logon and leave to make coffee, so I miss the prompt, which fails, so the Docker Desktop doesn't start.)

**Comment by ThePlenkov** [comment]
it disappeared with v.4.34.1

**Comment by hrabatom** [comment]
> it disappeared with v.4.34.1

Today I've got an update and I can confirm it's better now.
